flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/sink/DynamoDbTypeInformedElementConverter.java

```diff
@@ -0,0 +1,245 @@
+ */
+
+package org.apache.flink.connector.dynamodb.sink;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.typeinfo.BasicArrayTypeInfo;
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.NumericTypeInfo;
+import org.apache.flink.api.common.typeinfo.PrimitiveArrayTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeutils.CompositeType;
+import org.apache.flink.api.connector.sink2.SinkWriter;
+import org.apache.flink.api.java.tuple.Tuple;
+import org.apache.flink.api.java.typeutils.ObjectArrayTypeInfo;
+import org.apache.flink.api.java.typeutils.TupleTypeInfo;
+import org.apache.flink.connector.base.sink.writer.ElementConverter;
+import org.apache.flink.util.FlinkRuntimeException;
+
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link ElementConverter} that converts an element to a {@link DynamoDbWriteRequest} using
+ * TypeInformation provided.
+ */
+@PublicEvolving
+public class DynamoDbTypeInformedElementConverter<inputT>
+        implements ElementConverter<inputT, DynamoDbWriteRequest> {
+    private final CompositeType<inputT> typeInfo;
+
+    /**
+     * Creates a {@link DynamoDbTypeInformedElementConverter} that converts an element to a {@link
+     * DynamoDbWriteRequest} using the provided {@link CompositeType}. Usage: {@code new
+     * DynamoDbTypeInformedElementConverter<>(TypeInformation.of(MyPojoClass.class))}
+     *
+     * @param typeInfo The {@link CompositeType} that provides the type information for the element.
+     */
+    public DynamoDbTypeInformedElementConverter(CompositeType<inputT> typeInfo) {
+        this.typeInfo = typeInfo;
+    }
+
+    @Override
+    public DynamoDbWriteRequest apply(inputT input, SinkWriter.Context context) {
+        try {
+            return DynamoDbWriteRequest.builder()
+                    .setType(DynamoDbWriteRequestType.PUT)
+                    .setItem(convertElementUsingTypeInfo(input, typeInfo))
+                    .build();
+        } catch (IllegalArgumentException e) {
+            throw new FlinkRuntimeException("Couldn't convert Element to AttributeVal", e);
+        }
+    }
+
+    private <attT> Map<String, AttributeValue> convertElementUsingTypeInfo(
+            attT t, CompositeType<attT> typeInfo) {
+        Map<String, AttributeValue> map = new HashMap<>();
+        for (String fieldKey : typeInfo.getFieldNames()) {
+            TypeInformation<attT> fieldType = typeInfo.getTypeAt(fieldKey);
+            try {
+                Field field = t.getClass().getDeclaredField(fieldKey);
```

Comment:
As per the [coding guidelines](https://flink.apache.org/how-to-contribute/code-style-and-quality-java/#java-reflection) we should not use reflection. Can we instead use the `TypeExtractor` class as suggested in the docs? 

Another note here is that loading the class information for each record is expensive. Suggest that we lazily cache the type info based on the class.